compiler/src/dmd/statementsem.d

```diff
@@ -962,6 +962,12 @@ Statement statementSemanticVisit(Statement s, Scope* sc)
                     return retError();
                 }
 
+                if (dim == 2 && (*fs.parameters)[0].storageClass & STC.ref_)
+                {
+                    deprecation(fs.loc, "`%s` cannot be `ref`", (*fs.parameters)[0].toChars());
```

Comment:
explain to the user why it cannot be ref, e.g. "`foreach` iteration variable %s cannot be `ref`"

Reply:
Now done.